File: sdks/swift/Sources/Fireweave/Fireweave.swift
/// Fireweave SDK for Swift (spec v0.1.0).
///
/// Exactly two v1 capabilities (`spec/control-points.md` "Scope of v1"):
/// control points and target registration. Dependency budget: Foundation
/// only (`URLSession` for HTTP, `JSONSerialization`/`Codable` for JSON) —
/// `Package.swift`'s `dependencies:` array is empty.
///
/// ## The synchronous read surface
///
/// `controlPoints`'s nine methods are SYNCHRONOUS — a pure, lock-guarded
/// cache read (`Application/Runtime.swift`). `prefetch`/`initialize` are
/// `async` and populate that cache. This is the Phase 6 controller ruling
/// ("web's shape, not node's": a UI thread cannot `await` inside a render
/// path) — studied via `sdks/web`'s `ADR-0009` seam, built from
/// `spec/control-points.md` + `spec/modes.md` directly.
///
/// Quick start (local mode, offline):
///
/// ```swift
/// import Fireweave
///
/// let client = try await initFireweave(.local(InitFireweaveLocalOptions(
///     controlPoints: ["my-flag": true]
/// )))
/// client.controlPoints.getBooleanValue("my-flag", default: false)  // true, synchronously
/// await client.shutdown()
/// ```
///
/// There are no hidden global clients: everything is constructed explicitly
/// and injectable for tests.

/// Frozen SDK spec version this package implements (`spec/version.json`).
public let specVersion = "0.1.0"

File: sdks/dart/lib/fireweave.dart
/// Fireweave SDK for Dart (spec v0.1.0) — Flutter on every platform, the
/// Dart VM, and Dart compiled to JavaScript or WebAssembly.
///
/// Exactly two v1 capabilities (`spec/control-points.md` "Scope of v1"):
/// control points and target registration. Dependency budget: the Dart SDK
/// only (`dart:io` or the browser's `fetch` via `dart:js_interop` for HTTP,
/// `dart:convert` for JSON) — `pubspec.yaml` has no `dependencies:` block,
/// and no `flutter` SDK dependency either, so the same package runs in a
/// Flutter app on Android, iOS, macOS, Windows, Linux, and web, and in any
/// plain Dart program.
///
/// ## The synchronous read surface
///
/// `controlPoints`'s nine methods are SYNCHRONOUS — a pure cache read
/// (`FireweaveRuntime.evaluate`). `initFireweave`/`refresh`/`identify` are
/// `async` and populate that cache. This is "web's shape, not node's"
/// (ADR-0009, ADR-0011): a Flutter `build()` cannot `await`, so nothing on
/// the read path may suspend.
///
/// Quick start (local mode, offline):
///
/// ```dart
/// import 'package:fireweave/fireweave.dart';
///
/// final fw = await initFireweave(InitFireweaveOptions.local(
///   controlPoints: {'new-checkout': true},
/// ));
/// fw.controlPoints.getBooleanValue('new-checkout', false); // true, synchronously
/// await fw.shutdown();
/// ```
///
/// There are no hidden global clients: everything is constructed explicitly
/// and injectable for tests.
library;

export 'src/application/client.dart'
    show ControlPointsNamespace, ExtensionResult, FireweaveClient;
export 'src/application/init_fireweave.dart'
    show
        InitFireweaveLocalOptions,
        InitFireweaveOptions,
        InitFireweaveRemoteOptions,
        initFireweave;
export 'src/application/ports.dart'
    show
        AdapterResolution,
        ControlPointsBackendAdapter,
        EvaluateOptions,
        HttpTransport,
        LogSink,
        PrefetchOptions,
        PrefetchResult,
        RegisterTargetOptions,
        RegisterTargetResult,
        TransportResponse;
export 'src/application/runtime.dart'
    show
        FireweaveRuntime,
        LifecycleState,
        RuntimeConfig,
        defaultFlagsReadyTimeoutMs;
export 'src/domain/context.dart'
    show
        ContextLimits,
        EvaluationContext,
        allowedFireweaveContextKeys,
        defaultContextLimits,
        defaultReservedAttributeKeys,
        mergeContexts;
export 'src/domain/decision.dart' show Decision, DecisionReason;
export 'src/domain/errors.dart'
    show ErrorKind, FireweaveError, flagMetadataErrorKindKey, redactSecrets;
export 'src/domain/mode.dart' show Mode;
export 'src/domain/target.dart' show TargetKind;
export 'src/domain/types.dart'
    show
        FlagMetadata,
        FlagType,
        JsonObject,
        JsonValue,
        jsonEquals,
        matchesExpectedType,
        stableJsonString;
export 'src/domain/validation.dart'
    show
        Invalid,
        Valid,
        Validated,
        validateContext,
        validateControlPointKey,
        validateDefaultValue,
        validateInitOptions,
        validateTargetingKey;
export 'src/infrastructure/adapters/in_memory_adapter.dart'
    show FlagDefinition, InMemoryAdapter, InMemoryFault;
export 'src/infrastructure/adapters/local_adapter.dart'
    show FireweaveLocalAdapter, LocalRegisteredTarget;
export 'src/infrastructure/adapters/remote_adapter.dart'
    show FireweaveRemoteAdapter, RemoteAdapterConfig;
export 'src/infrastructure/hosts.dart'
    show assertHostAllowed, defaultAllowedHosts, isLoopbackHostname;

/// Frozen SDK spec version this package implements (`spec/version.json`).
const String specVersion = '0.1.0';
